jerry-core/debugger/jerry-debugger.c

```diff
@@ -0,0 +1,119 @@
+/* Copyright 2016 University of Szeged.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdbool.h>
+
+#include "jerry-debugger.h"
+#include "jerry-port.h"
+
+#define PORT 5001
+#define BACKLOG 1
+#define BUFFER_SIZE 128
+
+int sock;                     /**< return value of socket(), used other methods */
+int connected;                /**< return value of the whole socket connection */
+ssize_t byte_send;            /**< size of byte_send */
+size_t data_len;              /**< data lenght */
+bool optval = true;           /**< the arguments optval and optlen are used to access option values for setsockopt() */
+char* data[BUFFER_SIZE];      /**< data buffer */
```

Comment:
This is just char not char*. We are not string pointers here, only bytes. The current definition is 128 pointers which points to chars.